src/sage/combinat/partition.py

```diff
@@ -8784,14 +8930,198 @@ def cardinality(self):
         return ZZ(ans)
 
 
+##########################################
+# Partitions_length_and_parts_restricted #
+##########################################
+
+class Partitions_length_and_parts_restricted(Partitions):
+    r"""
+    The class of all integer partitions having parts and length in a
+    given range.
+
+    This class is strictly more general than
+    :class:`PartitionsGreatestLE`, except that we insist that the
+    size of the partition is positive and that neither the
+    restrictions on the parts not on the length are contradictory.
+
+    INPUT:
+
+    - ``n`` -- the size of the partition, positive
+    - ``min_length`` -- the lower bound on the number of parts, between 1 and n
+    - ``max_length`` -- the upper bound on the number of parts, between min_length and n
+    - ``min_part`` -- the bound on the smallest part, between 1 and n
+    - ``max_part`` -- the bound on the largest part, between min_part and n
+
+    EXAMPLES::
+
+        sage: from sage.combinat.partition import Partitions_length_and_parts_restricted
+        sage: Partitions_length_and_parts_restricted(10, 1, 10, 2, 5)
+        Partitions of 10 whose parts are between 2 and 5
+        sage: list(Partitions_length_and_parts_restricted(9, 3, 4, 2, 4))
+        [[4, 3, 2], [3, 3, 3], [3, 2, 2, 2]]
+
+        sage: [4,3,2,1] in Partitions_length_and_parts_restricted(10, 1, 10, 2, 10)
+        False
+        sage: [2,2,2,2,2] in Partitions_length_and_parts_restricted(10, 1, 10, 2, 10)
+        True
+
+    """
+    def __init__(self, n, min_length, max_length, min_part, max_part):
+        """
+        Initialize ``self``.
+
+        TESTS::
+
+            sage: from sage.combinat.partition import Partitions_length_and_parts_restricted
+            sage: p = Partitions_length_and_parts_restricted(10, 2, 5, 3, 4)
+            sage: TestSuite(p).run()
+        """
+        if not (1 <= min_part <= max_part <= n):
+            raise ValueError(f"min_part (={min_part}) and max_part (={max_part}) should satisfy 1 <= min_part <= max_part <= n (={n})")
+        if not (1 <= min_length <= max_length <= n):
+            raise ValueError(f"min_length (={min_length}) and max_length (={max_length}) should satisfy 1 <= min_length <= max_length <= n (={n})")
+        Partitions.__init__(self)
+        self._n = n
+        self._min_part = min_part
+        self._max_part = max_part
+        self._min_length = min_length
+        self._max_length = max_length
+
+    def _repr_(self):
+        """
+        Return a string representation of ``self``.
+
+        TESTS::
+
+            sage: from sage.combinat.partition import Partitions_length_and_parts_restricted
+            sage: Partitions_length_and_parts_restricted(9, 1, 9, 2, 9)
+            Partitions of 9 whose parts are at least 2
+            sage: Partitions_length_and_parts_restricted(9, 3, 5, 2, 9)
+            Partitions of 9 having length between 3 and 5 and whose parts are at least 2
```

Comment:
I think each type of conditions should be printed at least once (see the Codecov).